File: core/LongestUncommonSubsequenceII/LongestUncommonSubsequenceII.py
from typing import List


def findLUSlength(strs: List[str]) -> int:
    def is_seq(x, y):
        left, right = 0, 0
        while left < len(x) and right < len(y):
            if x[left] == y[right]:
                right += 1
            left += 1
        return right == len(y)

    res = -1
    for i in range(len(strs)):
        j = 0
        while j < len(strs):
            if i == j:
                j += 1
                continue

            if is_seq(strs[j], strs[i]):
                break

            j += 1

        if j == len(strs):
            res = max(res, len(strs[i]))

    return res


if __name__ == '__main__':
    print(findLUSlength(["aaa","aaa","aa"]))

File: core/LongestUncommonSubsequenceII/LongestUncommonSubsequenceII.java
package com.dbc;

public class LongestUncommonSubsequenceII {
    public boolean isSeq(String x, String y) {
        int index = 0;
        for (int i = 0; i < y.length() && index < x.length(); i++) {
            if (x.charAt(index) == y.charAt(i)) {
                index++;
            }
        }
        return index == x.length();
    }

    public int findLUSlength(String[] strs) {
        int res = -1;
        for (int i = 0, j; i < strs.length; i++) {
            for (j = 0; j < strs.length; j++) {
                if (i == j) continue;
                if (isSeq(strs[i], strs[j])) break;
            }
            if (j == strs.length) {
                res = Math.max(res, strs[i].length());
            }
        }
        return res;
    }
}
